app/src/main/java/org/mifos/mobile/ui/qr_code_display/QrCodeDisplayViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package org.mifos.mobile.ui.qr_code_display
+
+import android.graphics.Bitmap
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import org.mifos.mobile.utils.QrCodeGenerator
+
+class QrCodeDisplayViewModel: ViewModel() {
+
+    private var qrString: MutableStateFlow<String?> = MutableStateFlow(null)
+
+    val qrCodeDisplayUiState = qrString
+        .flatMapLatest { qrString ->
+            flow {
+                val qrBitmap = QrCodeGenerator.encodeAsBitmap(str = qrString)
+                if(qrBitmap == null) {
+                    emit(QrCodeDisplayUiState.Error())
+                } else {
+                    emit(QrCodeDisplayUiState.Success(qrBitmap = qrBitmap))
+                }
+            }
+        }.catch {
+            emit(QrCodeDisplayUiState.Error(errorMessage = it.message))
```

Comment:
Let's add a module in core:logs and define a method to register firebase non fatel error. On error send firebase crash error so we can know that this is working for everyone.

Reply:
updated